test/elixir_bench_web/schema/schema_test.exs

```diff
@@ -78,31 +80,416 @@ defmodule ElixirBenchWeb.SchemaTest do
+        jobs {
+          id
+        }
+      """
+
+      json_data = %{
+        "data" => %{
+           "jobs" => [
+             %{"id" => Enum.at(jobs_ids, 0)},
+             %{"id" => Enum.at(jobs_ids, 1)},
+             %{"id" => Enum.at(jobs_ids, 2)}
+           ]
+         }
+       }
+
+      {:ok, resp} =
+        context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      assert ^json_data = resp
+    end
+
+    test "return empty list when there is no registered jobs", context do
+      query = """
+        jobs {
+          id
+        }
+      """
+
+      {:ok, resp} =
+        context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      assert response_match?(:empty_data, resp, "jobs")
+    end
+  end
+
+  describe "job query" do
+    test "fetch job by id", context do
+      job = insert(:job)
+
+      query = """
+        job (id: "#{job.id}") {
+          id
+        }
+      """
+
+      json_data = %{
+        "data" => %{
+           "job" => %{
+             "id" => "#{job.id}",
+           }
+         }
+       }
+
+      {:ok, resp} =
+        context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      assert ^json_data = resp
+    end
+
+    test "return not found message when job does not exist", context do
+      query = """
+        job (id: 9999)
+        {
+          id
+        }
+      """
+
+      {:ok, resp} =
+        context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      error_message = Enum.at(resp["errors"], 0)
+
+      assert %{"message" => "not_found"} = error_message
+    end
+
+    test "return required field message when missing job fields", context do
+      required_fields = ["id"]
+
+      query = """
+        job {
+          id
+        }
+      """
+
+      resp_body =
+        context.conn
+        |> execute_query(query)
+        |> Map.get(:resp_body)
+
+      assert_required_fields(resp_body, required_fields)
+
+    end
+  end
+
+  defp execute_query(conn, query) do
+    conn
+    |> post("/api", AbsintheHelpers.query_skeleton(query))
+  end
+
+  defp decode_response_body(response) do
+    response
+    |> Map.get(:resp_body)
+    |> Antidote.decode
+  end
+
+  defp assert_required_fields(resp_body, required_fields) do
+    Enum.map(required_fields, fn field ->
+      assert(
+        response_match?(:required_field, resp_body, field),
+        "Error message not found for field \"#{field}\" in #{resp_body}"
+      )
+    end)
+  end
+
+  """
+  This function follows the Absinthe error message pattern and can break if
+  this pattern changes in future versions. Therefore it should be used inside
+  another assertive function like in assert_required_fields/2
+  """
+  defp response_match?(:required_field, response, field) do
+    response =~ ~r/In argument \\\"#{field}\\\": Expected type .*, found null/
+  end
+
+  defp response_match?(:empty_data, response, query) do
```

Comment:
not sure pattern matching on the first argument here to determine the type of work we want to do is our best call. The caller of the functions always needs to know explicitly which function to call - might as well be 2 different functions :)

Reply:
I created a module in test helpers for custom assertions, but maybe it would be better to put in another file inside support directory, like you did in benchee.